### PR TITLE
[XLA:GPU] Updated hlo_to_llvm_ir to also emit PTX and re-enable tests using it.

### PR DESCRIPTION
**Pull request by nouiz**
@sanjoy

### CODE REVIEW

File: tensorflow/compiler/xla/service/gpu/tests/reduction_vectorization_tile_fit_sm60.hlo [hunk 1]
```diff
@@ -0,0 +1,20 @@
+// RUN: hlo_to_llvm_ir --ptx --sm=60 %s | FileCheck %s
```
Comment (sanjoy) [inline tensorflow/compiler/xla/service/gpu/tests/reduction_vectorization_tile_fit_sm60.hlo:1]: Can you please use the `// -----` separator to combine some of these tests into a single file, like in [here](https://github.com/tensorflow/tensorflow/blob/ddeb3ab137a3b8d8019582ee93ac3ed7f403cc26/tensorflow/compiler/xla/service/gpu/tests/scatter.hlo#L71)?

**Review by sanjoy — CHANGES_REQUESTED** [review]

File: tensorflow/compiler/xla/service/gpu/tests/reduction_vectorization_tile_fit_sm60.hlo (hunk 1 above)
Comment (nouiz) [reply tensorflow/compiler/xla/service/gpu/tests/reduction_vectorization_tile_fit_sm60.hlo]: I was looking for a way to reduce the number of file. Thanks fo the guidance.
It is done.

File: tensorflow/compiler/xla/service/gpu/nvptx_compiler.h [hunk 2]
```diff
@@ -32,6 +32,8 @@ namespace gpu {
 
 void WarnIfBadDriverJITVersion();
 
+string GetLibdeviceDir(const HloModuleConfig& hlo_module_config);
```
Comment (sanjoy) [inline tensorflow/compiler/xla/service/gpu/nvptx_compiler.h:35]: Please add a comment.

**Review by sanjoy — CHANGES_REQUESTED** [review]
Minor nit.

File: tensorflow/compiler/xla/service/gpu/nvptx_compiler.h (hunk 2 above)
Comment (nouiz) [reply tensorflow/compiler/xla/service/gpu/nvptx_compiler.h:36]: Done.

**Review by sanjoy — APPROVED** [review]

**Comment by gbaned** [comment]
@nouiz  Can you please address Ubuntu Sanity errors? Thanks!

**Comment by nouiz** [comment]
Done. I also updated the new tests. I had forgot the CHECK-LABEL.

**Review by sanjoy — APPROVED** [review]

**Review by gbaned — APPROVED** [review]

**Comment by nouiz** [comment]
I do not see the new commit in the master branch:
```
commit 979cabfc2df277a143c88303f78d172374358ad4 (origin/upstream-hlo_to_llvm_ir, upstream-hlo_to_llvm_ir)
Author: Frederic Bastien <fbastien@nvidia.com>
Date:   Tue Mar 9 08:13:12 2021 -0800

    Added check for SM 70.

commit 3bb586d99c375b2bd0a1d285e616d366e7da6c9d
Author: Frederic Bastien <fbastien@nvidia.com>
Date:   Tue Mar 9 08:04:08 2021 -0800

    Simplify the check.

commit 57b2b07fd97fef115d6e3618384d620b3b41c4d7
Author: Frederic Bastien <fbastien@nvidia.com>
Date:   Tue Mar 9 07:59:11 2021 -0800

    Combine tests into a single file.

commit 482a34c6ae000f7154a145cfcb8889523ccba750
Author: Frederic Bastien <fbastien@nvidia.com>
Date:   Tue Mar 9 07:48:11 2021 -0800

    Combine tests into a single file.

commit 32de6bb747b092265a70d75430d40a57307b399a
Author: Frederic Bastien <fbastien@nvidia.com>
Date:   Tue Mar 9 07:39:05 2021 -0800

    Combine tests into a single file.
```

Do you need to include them manually?

**Comment by sanjoy** [comment]
You'll have to open a new PR, your original change has already been merged at bcf9c308e231d90721a064460e9fbe3287131f54.

**Comment by nouiz** [comment]
https://github.com/tensorflow/tensorflow/pull/47707

**Pull request closed without merging**